promptExecutor/src/main/java/explorableviz/transparenttext/TestQueryContext.java

```diff
@@ -41,7 +40,16 @@ public static TestQueryContext importFromJson(JSONObject testCase, Random random
         JSONObject json_variables = testCase.getJSONObject("variables");
         JSONArray json_imports = testCase.getJSONArray("imports");
         String code = testCase.getString("code");
-        String file = testCase.getString("text");
+        JSONArray paragraph = testCase.getJSONArray("paragraph");
+        StringBuilder text = new StringBuilder();
+        for(int i = 0; i < paragraph.length(); i++) {
+            JSONObject paragraph_element = paragraph.getJSONObject(i);
+            if(paragraph_element.getString("type").equals("string")) {
```

Comment:
I’m going to merge this PR anyway so we can keep things moving but I suggest rewriting this as:

```java
        if (paragraph_element.getString(“type”).equals(“string”)) {
            text.append(paragraph.getJSONObject(i).getString(“value”));
        } else
        if (paragraph_element.getString(“type”).equals(“expr”)) {
            text.append(paragraph.getJSONObject(i).getString(“expr”));
        } else {
            throw new RuntimeError(“Unexpected element type”)
        }
```
because that’s a tighter contract and will fail with a useful error message if the file doesn’t match the expected format.